ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_miscellaneous.py

```diff
@@ -370,6 +370,49 @@ def test_numpy_gcd(
     )
 
 
+# gradient
+@handle_frontend_test(
+    fn_tree="numpy.gradient",
+    dtype_input_axis=helpers.dtype_values_axis(
+        available_dtypes=("float32", "float16", "float64"),
+        min_num_dims=1,
+        max_num_dims=3,
+        min_dim_size=2,
+        max_dim_size=4,
+        valid_axis=True,
+        force_int_axis=True,
+    ),
+    varargs=helpers.ints(
+        min_value=-3,
+        max_value=3,
+    ),
```

Comment:
the `varargs` argument can be one of the following as per the function documentation:
1. single scalar to specify a sample distance for all dimensions.
2. N scalars to specify a constant sample distance for each dimension.
       i.e. `dx`, `dy`, `dz`, ...
3. N arrays to specify the coordinates of the values along each
       dimension of F. The length of the array must match the size of
       the corresponding dimension
4. Any combination of N scalars/arrays with the meaning of 2. and 3.

so please extend the testing strategy to accommodate these cases

Reply:
okay, i'm working on it. thank you